File: 2785.cpp
struct Solution {   
    string sortVowels(string &s) {
        int counter[10] = {0};
        vector<int> indices;
        const string refStr = "AEIOUaeiou";
        for (int i = 0; i < s.size(); ++i){
            char c = s[i];
            if ((0x208222 >> (c & 31)) & 1){ // is vowel
                counter[refStr.find(c)] += 1;
                indices.push_back(i);
            }
        }
        int ptr = 0;
        for (int idx : indices){
            while (counter[ptr] == 0){
                ++ptr;
            }
            s[idx] = refStr[ptr];
            --counter[ptr];
        }
        return s;
    }
};

File: sol/2785/2785.py
class Solution:
    def sortVowels(self, s: str) -> str:
        res = list(s)
        refStr = 'AEIOUaeiou'
        refSet = frozenset(refStr)
        vowelCount = [0] * 10
        vowelIndices = list()
        for i, c in enumerate(s):
            if c in refSet:
                vowelIndices.append(i)
                vowelCount[refStr.index(c)] += 1
        ptr = 0
        for idx in vowelIndices:
            while vowelCount[ptr] == 0:
                ptr += 1
            res[idx] = refStr[ptr]
            vowelCount[ptr] -= 1
        return ''.join(res)

